tidy3d/plugins/adjoint/web.py

```diff
@@ -114,3 +146,180 @@ def run_bwd(
 
 # register the custom forward and backward functions
 run.defvjp(run_fwd, run_bwd)
+
+
+""" Running a batch of simulations using web.run_async. """
+
+
+def _task_name_orig(index: int, task_name_suffix: str = None):
+    """Task name as function of index into simulations. Note: for original must be int."""
+    if task_name_suffix is not None:
+        return f"{index}{task_name_suffix}"
+    return int(index)
+
+
+# pylint:disable=too-many-locals
+@partial(custom_vjp, nondiff_argnums=tuple(range(1, 6)))
+def run_async(
+    simulations: Tuple[JaxSimulation, ...],
+    folder_name: str = "default",
+    path_dir: str = DEFAULT_DATA_DIR,
+    callback_url: str = None,
+    num_workers: int = None,
+    task_name_suffix: str = None,
+) -> Tuple[JaxSimulationData, ...]:
+    """Submits a set of :class:`.JaxSimulation` objects to server, starts running,
+    monitors progress, downloads, and loads results
+    as a tuple of :class:`.JaxSimulationData` objects.
+    Uses ``asyncio`` to perform these steps asynchronously.
+
+    Parameters
+    ----------
+    simulations : Tuple[:class:`.JaxSimulation`, ...]
+        Mapping of task name to simulation.
+    folder_name : str = "default"
+        Name of folder to store each task on web UI.
+    path_dir : str
+        Base directory where data will be downloaded, by default current working directory.
+    callback_url : str = None
+        Http PUT url to receive simulation finish event. The body content is a json file with
+        fields ``{'id', 'status', 'name', 'workUnit', 'solverVersion'}``.
+    num_workers: int = None
+        Number of tasks to submit at once in a batch, if None, will run all at the same time.
+
```

Comment:
oh right, so this is actually just for internal purposes for appending `_adj` and `_fwd` to the task names.
The user is actually not supposed to use this as jax can't handle differentiating a dictionary with `str` keys, so in this `adjoint.web_async` function the sims are passed as a list and the task names constrained to be `int` index into the sims_list. 

So I will just leave this docstring out given that if the user supplies it, it will not work properly and they'll get some 500 line jax error :D